lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -25,7 +25,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
   serialize: function(_model) {
     var model = _model || this.context,
         names = this.names,
-        handler = this.handler;
+        handler = this.handler,
+        serializer = this.serializer || (handler && handler.serialize);
```

Comment:
I think that the combination of lazy-loading with Route#serialize leads to inconsistent behavior that can lead to hard to find bugs, that's why I think it would be better not to support this combination by allowing for a way to disable the use of Route#serialize.